arch/risc-v/src/mpfs/mpfs_coremmc.c

```diff
@@ -2063,6 +2064,20 @@ static sdio_eventset_t mpfs_eventwait(struct sdio_dev_s *dev)
         {
           /* Yes... break out of the loop with wkupevent non-zero */
 
+#ifdef CONFIG_BUILD_FLAT
```

Comment:
I don't think we should do this. If there is sleeping / re-try in some form, it should be done in upper levels in the stack, after receiving -EBUSY from mpfs_sendcmd

Reply:
Actually; I just realized that there is always a way to check if the card is busy writing.

If the coremmc block conforms to the SD physical layer spec, the controller should send out busy tokens to the data lines as long as it is busy writing.  This means that you should be able to trigger on interrupt on data0 line, after all the data has been sent to the card.

Maybe we could wire the data0 line out from the card / and or connect it on the fpga also to some GPIO to be able to monitor the write status on the card?

Reply:
Yes, to achieve the maximum performance without polling, the way is really to study the WRCOMPLETE thing I pointed out above. It should always be possible to trigger on interrupt when the write is completed

Reply:
I found the reason why the transferready did not work properly. The mpfs_sendcmd waits for data lines ready until sending the command... So it prevents mmcsd driver to poll card busy status, but hangs in busyloop.
Just removing the mpfs_check_lines_busy() call from mpfs_sendcmd() fixed the problem.